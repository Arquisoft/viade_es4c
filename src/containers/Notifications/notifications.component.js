import React,{useState,useCallback,useEffect} from "react";
import {NotificationsList} from "./children";
import {ldflexHelper, errorToaster} from "../.././utils";

const Notifications = (props) => {
    const [inboxes, setInbox] = useState([]);
    const { webId } = props;
    
    const discoverInbox = useCallback(async () => {
        try {
          let inboxes = [];
          /**
           * Get user's global inbox path from pod.
           */
          const globalInbox = await ldflexHelper.discoverInbox(webId);
    
          if (globalInbox) {
            inboxes = [
              ...inboxes,
              { path: globalInbox, inboxName: "Global", shape: "default" }
            ];
          }
          /**
           * If user doesn't has inbox in his pod will show an error and link to
           * know how fix it.
           */
          if (inboxes.length === 0) {
            errorToaster("Your inbox couldn't be found","Error");
          }
          setInbox(inboxes);
        } catch (error) {
          console.error(error);
          errorToaster("An error has occurred with the inbox folder","Error");
        }
      }, [webId]);

      const inboxUrl = inboxes.map((item) => item.path);

      useEffect(() => {
        if (webId) {
          discoverInbox();
        }
      }, [webId,discoverInbox]);

  return (
    <div>
      <NotificationsList inboxes={inboxUrl} {...props}/>
      </div>
  );
};

export default Notifications;
