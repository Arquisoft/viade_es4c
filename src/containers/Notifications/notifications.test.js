import React from 'react';
import { shallow } from 'enzyme';
import Notifications from "./notifications.component";

describe('Notifications', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Notifications debug />);
  
    expect(component).toMatchSnapshot();
  });
});
