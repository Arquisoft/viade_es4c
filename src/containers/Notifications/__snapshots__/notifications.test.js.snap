// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications should render correctly in "debug" mode 1`] = `
<div
  style={
    Object {
      "margin": "50px",
    }
  }
>
  <h1>
    Notifications
  </h1>
  <NotificationsList
    debug={true}
    inboxes={Array []}
  />
</div>
`;
